src/server/command_registry.h

```diff
@@ -169,7 +169,15 @@ class CommandRegistry {
   CommandRegistry& operator<<(CommandId cmd);
 
   const CommandId* Find(std::string_view cmd) const {
-    auto it = cmd_map_.find(cmd);
+    const auto it = cmd_map_.find(cmd);
+    if (it == cmd_map_.end()) {
+      if (const auto alias_it = cmd_aliases_.find(cmd); alias_it != cmd_aliases_.end()) {
+        if (alias_it->first == alias_it->second) {
+          return nullptr;
+        }
+        return Find(alias_it->second);
+      }
+    }
```

Comment:
I don't have a strong opinion about this but if you want to avoid the nested ifs you can reverse the condition:

```
    if (it != cmd_map_.end()) {
      return &it->second;
    }
    if (const auto alias_it = cmd_aliases_.find(cmd); alias_it != cmd_aliases_.end()) {
      if (alias_it->first == alias_it->second) {
          return nullptr;
      }
      return Find(alias_it->second);
    }
    return nullptr;
```

I truly don't care just mentioning it here as a way to get of multiple nested ifs

Reply:
thanks, changed, it looks easier to understand with early return

Reply:
"easier" is a strong word, whatever works for you as well especially for these kind of suggestions :smile: